### ISSUE TITLE
Install file tells users that opm-dune-cmake module is mandatory.

### ISSUE THREAD

**Issue by blattms** [OPM/ewoms#261]
Stumbled over this while looking at #260. 

[Install file line 18](/OPM/ewoms/blob/80ebbe25c2c71bfa58b7bff1f02b7b921263649f/INSTALL#L18-L21) ff reads
```
In order to compile eWoms, you first need to install the following OPM
and DUNE modules:

- opm-dune-cmake from [3]
```
und in [line 124](/OPM/ewoms/blob/80ebbe25c2c71bfa58b7bff1f02b7b921263649f/INSTALL#L18)
```
[3] https://github.com/dr-robertk/opm-dune-cmake
```

At least the last time I used ewoms this was not needed. Yet the file is phrased like using opm-dune-cmake is mandatory. Seems a bit unfortunate. Maybe it would be a good idea to rephrase this as another possible option and mention the concrete use case.

BTW: Isn't dune-localfunctions optional, too?

**Comment by andlaus** [comment]
Well, these are the installing instructions for the eWoms module, and while opm-dune-cmake is not mandatory in a technical sense I *strongly* recommend using it. To see the point, try to compile it with Dune 2.6 and the default build system. IOW, if you don't want to follow the recommendation, you're free to do so, but don't expect any support from my side. (also, if you don't like the wording of `INSTALLING`, open a PR.)

dune-localfunctions is basically the same: yes, it is technically optional but it is needed for the vertex-centered FV discretion with something more sophisticated than TPFA. Some people will want to play round with this and installing dune-localfunctions in addition to the rest is not a big hassle.

**Comment by akva2** [comment]
what issues did you have building against dune 2.6 rc1? everything built completely hassle-free for me, there was a build issue in opm-simulators, due to changes in dune-istl API (trivial to resolve). and a couple of similarly caused build issue in opm-upscaling. but i had no issues related to the build system.

please tell so i can fix.

**Comment by andlaus** [comment]
The PRs which were needed for errors in the C++ library parts have been merged over the course of last week. There are still plenty of deprecation warnings in opm-grid and even some errors when compiling the opm-grid unit tests (at least not with NDEBUG defined).

With the default BS, I still suffer from OPM/opm-common#296, dune-alugrid and dune-fem don't work, and last time I tested there were issues with SuperLU and/or UMFPack. Note that this are obviously only the packages that I've tried.

**Comment by akva2** [comment]
None of what you list is related to dune 2.6 issues in the build system, so still waiting.

**Comment by andlaus** [comment]
> None of what you list is related to dune 2.6 issues in the build system.

This statement is probably technically correct, but you were asking what about build issues when using Dune 2.6, not for issues which have are exclusively caused by Dune 2.6. Note that, for the vast majority of people, this is a mood point simply because the build fails and they won't care who's fault it is. Further, I suppose that these errors also occur if the Dune packages of newer debian based distributions are used, because these  distributions ship Dune 2.5 and Dune 2.5 fails for me as well. (Dune 2.4 works, BTW.)

Since there is a working solution which I can recommend with a straight face, let's wind that discussion down. Please?

**Comment by alfbr** [comment]
> I still suffer from OPM/opm-common#296

How do you suffer from this?

**Comment by akva2** [comment]
jenkins uses dune packages on newest debian stable. jenkins builds against suitesparse on newest debian stable.if things were broken for the vast majority of people, i would know.
i refuse to wind down the discussion just because you were arrested throwing false accusations around.

**Comment by joakim-hove** [comment]
> How do you suffer from this?

![image](https://user-images.githubusercontent.com/1321665/34869586-a9c8e922-f787-11e7-991d-84da2d41d7ca.png)

**Comment by andlaus** [comment]
>> I still suffer from OPM/opm-common#296
>
> How do you suffer from this?

I still get the linker errors I described in OPM/opm-common#297. anyway, I'm happy, so let's not argue about it. 

> jenkins uses dune packages on newest debian stable.

dune 2.5? this probably means that 2.4 is not tested by jenkins anymore?

> if things were broken for the vast majority of people, i would know.

For a user it does not matter if it does work for the vast majority of people if it does not work for him.

> i refuse to wind down the discussion just because you were arrested throwing false accusations around.

these "accusations" are consistent with my experience, though. If you send me your SSH public key, you could look yourself.

**Comment by akva2** [comment]
you can include dune2.4 in the trigger and it will build with dune 2.4 in addition. and i build against 2.4 on my xenial all the time. 

i don't doubt that you see what you see, nor do i even doubt there are issues, but you need to start behaving like a grown up and ditch the everything-is-a-conflict attitude. all i ask is that you show some respect, and stop throwing terms like 'bs' around.

dune-istl 2.6 expects some new defines for superlu to work, so that does indeed need some work, which i have done and will open a pr shortly. this is caused by template libraries, who by their very nature breaks the first rule of software library development (encapsulation).

**Comment by alfbr** [comment]
> I still get the linker errors I described in OPM/opm-common#297

Well none of us have been able to reproduce it. I even did give you my ssh key to go check, only to find out that your set-up did not build at all, and it had nothing to do with linker errors. It was your build system that was broken. Frankly, if you claim that the current build system has issues, and you even claim to know what those issues are, we should at least be given enough information to reproduce it.

> dune 2.5? this probably means that 2.4 is not tested by jenkins anymore?

Well, I have a jenkins instance here compiling against 2.4 on a daily basis. Performance monitoring on PTS is also compiling and testing performance against 2.4 daily. If you believe the current testing is insufficient, then please be explicit.

**Comment by andlaus** [comment]
> only to find out that your set-up did not build at all,

why have you stayed silent for several weeks then? anyway, I had a look at what you did and it seems like you did not even try the default build system and the problem using DUNE was caused by some screw up in opm-upscaling (which should be only of limited interest in that context)

**Comment by andlaus** [comment]
closed via #270

**Comment by alfbr** [comment]
> why have you stayed silent for several weeks then?

What silence? The first attempt I did you had removed my ssh key before I could test. The second attempt I reported to you immediately that you set-up was broken. Moreover, in your answer you claimed that this fixed it:
https://github.com/OPM/opm-core/pull/1200
but warned me that my ssh key was removed again.
> I had a look at what you did and it seems like you did not even try the default build system

Oh, so you do recall me not being silent after all.
